datadog/resource_datadog_monitor.go

```diff
@@ -383,7 +386,14 @@ func resourceDatadogMonitorRead(d *schema.ResourceData, meta interface{}) error
 	d.Set("name", m.GetName())
 	d.Set("message", m.GetMessage())
 	d.Set("query", m.GetQuery())
-	d.Set("type", m.GetType())
+	if typ, ok := m.GetTypeOk(); ok {
+		if d.Get("type").(string) == "metric alert" && typ == "query alert" {
```

Comment:
If we do this here, do we still need the DiffSuppress in the schema?

Reply:
Good question, I'm not sure. I'll find out :)

Reply:
You were right, `DiffSuppressFunc` is no longer necessary with this. I added a comment explaining it and removed `DiffSuppressFunc`.